README.md

```diff
@@ -1,2 +1,31 @@
 # comp110-worksheet-5
 Base repository for COMP110 worksheet 5
+
+
+## A)
+Checks every value within the list against every value within the list to determine if there were any duplicates within the list.
+If there is a duplicate within the list it will return true, if not then it will return false.
+
+## B)
+The  first for loop is done until the value of N in the worst case scenario, the second for loop is also done until the value of N. Therefore, technically this is n x n, which is the same as being quadratic. O(n²)
+
+## C)
+The algorithm would still be correct as it will still check the whole list in order to find duplicates, the only difference is its using the other index. As there is no change to the rest of the code it would still return true.
+
+## D)
+Only half of operations are actually happening due to the fact that there are less comparisons happening with the change.
+
+## E)
+The time complexity of the algorithm would still remain quadratic. All values within the list are still checked.
+
+## F)
+It is called TimSort and has a time complexity of: O(nlogn) (https://dev.to/s_awdesh/timsort-fastest-sorting-algorithm-for-real-world-problems--2jhd)
+
+## G)
+O(n) it should be Linear time complexity because every value in the list is being checked using the i index.
```

Comment:
However the time for the sort must also be taken into account, so it would be `O(n log n + n) = O(n log n)`